oops/backplane/gold_master.py

```diff
@@ -0,0 +1,2112 @@
+                    for key in keys:
+                        sum = np.sum(self.backplane.intercepts[i][key].mask)
+                        LOGGING.diagnostic('   ', key, sum)
+
+                LOGGING.diagnostic('\nGridless arrivals')
+                keys = list(self.backplane.gridless_arrivals.keys())
+                keys.sort(key=BackplaneTest._sort_key)
+                for key in keys:
+                    sum = np.sum(self.backplane.gridless_arrivals[key].mask)
+                    LOGGING.diagnostic('   ', key, sum)
+
+                LOGGING.diagnostic('\nBackplanes')
+                keys = list(self.backplane.backplanes.keys())
+                keys.sort(key=BackplaneTest._sort_key)
+                for key in keys:
+                    sum = np.sum(self.backplane.backplanes[key].mask)
+                    if key in self.backplane.backplanes_with_derivs:
+                        derivs = self.backplane.backplanes_with_derivs[key].derivs
+                        flag = ' '
+                        if 't' in derivs:
+                            if 'los' in derivs:
+                                flag = '*'
+                            else:
+                                flag = 't'
+                        elif 'los' in derivs:
+                            flag = 'l'
+                    else:
+                        flag = ' '
+                    LOGGING.diagnostic('   %s%s' % (flag, key), sum)
+
+                LOGGING.diagnostic('\nAntimasks')
+                keys = list(self.backplane.antimasks.keys())
+                keys.sort()
+                for key in keys:
+                    antimask = self.backplane.antimasks[key]
+                    info = ('array' if isinstance(antimask, np.ndarray)
+                                    else str(antimask))
+                    LOGGING.diagnostic('   ', key, '(%s)' % info)
+
+                LOGGING.diagnostic()
+                LOGGING.pop()
+
+            seconds = (datetime.datetime.now() - start).total_seconds()
+            LOGGING.info('Elapsed time: %.3f s' % seconds)
+
+        # Be sure to remove the BackplaneTest-specific file handler afterward
+        finally:
+            if LOGGING.warnings:
+                LOGGING.debug('Total warnings = ' + str(LOGGING.warnings))
+            if LOGGING.errors:
+                LOGGING.info('Total errors = ' + str(LOGGING.errors),
+                             force=True)
+
+            if self.args.log:
+                LOGGING.logger.removeHandler(handler)
+                handler.close()
+
+            LOGGING.pop()
+
+    @staticmethod
+    def _sort_key(key):
+        """Sort key function, needed to handle occurrences of Frames, Paths,
+        and None is some dictionary keys.
+
+        Also allow sorting among numbers, strings and tuples: numbers first,
+        strings second, objects third, tuples fourth.
+        """
+        if isinstance(key, (tuple, list)):
```

Comment:
Interesting. I had not seen this before. However, strings are also instances of collections.abc.Sequence, so I don't think this would improve clarity. This is a function for sorting event keys and intercept keys, and we have complete control over what they can contain, so I prefer to leave this as is.